src/main/java/ooo/oshi/foreign/windows/WtsApi32.java

```diff
@@ -0,0 +1,63 @@
+package ooo.oshi.foreign.windows;
+
+import java.lang.foreign.*;
+import java.lang.invoke.MethodHandle;
+
+import static java.lang.foreign.ValueLayout.*;
+
+public class WtsApi32 {
+
+    private static final SymbolLookup WTSAPI32;
+    private static final Linker LINKER;
+
+    static {
+        LINKER = Linker.nativeLinker();
+        System.loadLibrary("wtsapi32");
+        WTSAPI32 = SymbolLookup.loaderLookup();
+    }
+
+    private static MethodHandle methodHandle(String methodName, FunctionDescriptor fd) {
+        return LINKER.downcallHandle(WTSAPI32.lookup(methodName).orElseThrow(), fd);
+    }
+
+    private static final MethodHandle WTSEnumerateProcessEx = methodHandle("WTSEnumerateProcessesA", FunctionDescriptor.of(JAVA_BOOLEAN, JAVA_INT, JAVA_INT, JAVA_INT, ADDRESS, ADDRESS));
+
+    private static final MemoryLayout _WTS_PROCESS_INFOA = MemoryLayout.structLayout(
+        JAVA_INT.withName("SessionId"),
+        JAVA_INT.withName("ProcessId"),
+        JAVA_CHAR.withName("pProcessName"),
+        ADDRESS.withName("pUserSid")
+    ).withName("WTS_PROCESS_INFOA");
+
+    /**
+     * Refer to <a href="https://github.com/VFPX/Win32API/blob/master/libraries/wtsapi32/WTSEnumerateProcesses.md">Win32API</a>
+     */
+    public static void enumerateProcesses() {
+        try {
+            try (MemorySession session = MemorySession.openConfined()) {
+                int hServer = 0; // represents null handle
+                int version = 1; // version of enumerated request, must be 1
+                // out variables
+                Addressable ppProcessInfo = session.allocate(_WTS_PROCESS_INFOA); // array of WTS_PROCESS_INFO
```

Comment:
This seems like a better approach than #16 , have access to entire array.

ATM playing around with this and thinking `session.allocate(ADDRESS)` instead of struct here then after `pCount` is evaluated can somehow transform this into `MemoryLayout.sequenceLayout(pCount, _WTS_PROCESS_INFOA)` to make this an array